packages/server/src/GraphQLCache.js

```diff
@@ -411,23 +414,50 @@ export class GraphQLCache {
   ): Promise<?GraphQLSchema> => {
     const projectConfig = this._graphQLConfig.getProjectConfig(appName);
 
-    if (!projectConfig || !projectConfig.schemaPath) {
+    if (!projectConfig) {
       return null;
     }
 
     const projectName = appName || 'undefinedName';
-
     const schemaPath = projectConfig.schemaPath;
-    const schemaCacheKey = `${schemaPath}:${projectName}`;
+    const endpointInfo = this._getDefaultEndpoint(projectConfig);
+
+    let schemaCacheKey = null;
+    let schema = null;
 
-    if (this._schemaMap.has(schemaCacheKey)) {
-      const schema = this._schemaMap.get(schemaCacheKey);
-      return schema && queryHasExtensions
-        ? this._extendSchema(schema, schemaPath, projectName)
-        : schema;
+    if (endpointInfo) {
+      schemaCacheKey = `${endpointInfo.endpointName}:${projectName}`;
+
+      // Maybe use cache
```

Comment:
One thing that would be cool to do later is to read the schema info from the endpoint every time to invalidate the cache. When it was a physical `.graphql` file, the language service was able to detect the changes to the schema file as one of the extensions and extend the schema with it as well as invalidating it. I'd say it's pretty important to get the fresh state of the schema as long as we're editing the graphql files that depend on this schema.

A couple things about this:
- definitely not required to merge this PR
- it shouldn't probably refetch for every updates that triggers this method

Reply:
Agreed. I think periodic cache invalidation is probably the way to go.